### ISSUE TITLE
Run model error: Struct contents reference from a non-struct array object.

### ISSUE THREAD

**Issue by justbennet** [NISOx-BDI/SwE-toolbox#68]
I am getting this error when I try to run a saved model provided me by a researcher.

```
------------------------------------------------------------------------
Running job #1
------------------------------------------------------------------------
Running 'Run Model'
Initialising parameters                 :                        ...done
Output images                           :                  ...initialised
Plane   1/68 , block   1/1              :            ...read & mask dataFailed  'Run Model'
Struct contents reference from a non-struct array object.
In file "/usr/local/matlab/SwE-toolbox/swe_cp_WB.m" (???), function "swe_mask_seperable" at line 2009.
In file "/usr/local/matlab/SwE-toolbox/swe_cp_WB.m" (???), function "swe_cp_WB" at line 767.
In file "/usr/local/matlab/SwE-toolbox/swe_cp.m" (???), function "swe_cp" at line 61.
In file "/usr/local/matlab/SwE-toolbox/swe_run_rmodel.m" (???), function "swe_run_rmodel" at line 22.

The following modules did not run:
Failed: Run Model
```
I can provide the SwE.mat file and the generated error file if that would help.

**Comment by justbennet** [comment]
Sorry, this is using Matlab R2016b running on Ubuntu 16.04 LTS.

Is there a way to dump the SwE.mat file to text?  A way to read that back would also be very nice for those of us who help people with the software but aren't familiar enough with the analysis to be able to use the GUI well.

Thanks.

**Comment by nicholst** [comment]
Hi @justbennet

I fear this could be an incompatibility we've introduced with the new
version.  We weren't very careful a few times this year and made revisions
that broke SwE.mat compatibility.  We've now gotten our act together and
now all releases will be backwards compatible *unless* designated by an
major number change (which is the case for the 2.0.0 release candidate
we're working on
https://github.com/NISOx-BDI/SwE-toolbox/releases/tag/v2.0.0-rc ).

Can you see if you can reproduce this with a newly created job?

**Comment by justbennet** [comment]
Tom,

The SwE.mat file was generated using 1.2.10 or 1.2.11 version.

I have installed now from the current HEAD.  Is that what you would like me to get tested?

I will have to ask the researcher to create a new design and share it with me, and I can do so, I just want to make sure I have the version you'd like me to test checked out.  If you specifically want me to checkout a branch corrsponding to the v2.0.0-rc tag, I can do that.

**Comment by nicholst** [comment]
Let's not have to many changes flying around... I'd keep working with master and avoid the v2.0.0-rc for now.

**Comment by TomMaullin** [comment]
Hi @justbennet ,

Thank you for reporting this! Please could you send us your `SwE.mat` (maybe upload it to [GitHub Gists ](https://gist.github.com/)and send us a link if that is okay?) and just quickly confirm which version of the code you were running when you had the error (which branch you cloned/which release you downloaded, if possible which commit - you can check this by typing `git show` in the commandline if you are git fluent! But don't worry if not!). I will double check whether this is an known incompatibility issue between versions or something more severe.

(Also if you want to read back the SwE file itself, you can go into Matlab, change into the directory the file is stored in and type `load('SwE.mat')` to probe it directly. It will then appear as a structure named `SwE`. I hope this helps!)

**Comment by justbennet** [comment]
@TomMaullin I don't see a way to upload a binary file to a Gist.  I put the SwE.mat file at
http://www.umich.edu/~bennet/SwE.mat
for you.  Please let me know when you have a copy, and I'll delete it.

**Comment by justbennet** [comment]
@TomMaullin I reran the SwE.mat file just uploaded.  This is with
```
[bennet:mistakenhat SwE-toolbox]$ git show | head -1
commit d29829e0e1128b5354805f9a93d2d43229ee53bd
```
which has
```
remote.origin.url=https://github.com/NISOx-BDI/SwE-toolbox.git
```
I also uploaded the saved errors to http://www.umich.edu/~bennet/RunTimeErrors.mat in case that is useful.

The current errors save there and using the SwE.mat file provided are
```
>> addpath('/usr/local/MethodsCore/SPM/SPM12/spm12_with_R7219')
>> addpath('/usr/local/matlab/SwE-toolbox')
>> swe
   ___          ___           _      ___     _  _        
  / __) _    _ | __)   _  _  /_|    (__ \   /_|/_|       
  \__ \ \\/\// | __)   \\//   || _  / __/ _  || ||       
  (___/  \/\/  |___)    \/    |||_| \___)|_| || ||       

  swe v1.2.11 
Warning: MATLAB has disabled some advanced graphics rendering features by switching to software OpenGL.
For more information, click here. 
SwE present working directory:
	/nfs/turbo/heitzeg-turbo/researchers/lcope/DualProcessModels/DualProcess_SandwichEstimator/GNG/SwE_v1211_test/CRvBaseline/All/MainEffects


------------------------------------------------------------------------
Running job #1
------------------------------------------------------------------------
Running 'Run Model'
Initialising parameters                 :                        ...done
Output images                           :                  ...initialised
Plane   1/68 , block   1/1              :            ...read & mask dataFailed  'Run Model'
Dot indexing is not supported for variables of this type.
In file "/usr/local/matlab/SwE-toolbox/swe_cp_WB.m" (???), function "swe_mask_seperable" at line 2009.
In file "/usr/local/matlab/SwE-toolbox/swe_cp_WB.m" (???), function "swe_cp_WB" at line 767.
In file "/usr/local/matlab/SwE-toolbox/swe_cp.m" (???), function "swe_cp" at line 61.
In file "/usr/local/matlab/SwE-toolbox/swe_run_rmodel.m" (???), function "swe_run_rmodel" at line 22.

The following modules did not run:
Failed: Run Model
```

**Comment by justbennet** [comment]
I found the example on the NISOx web site

    http://www.nisox.org/Software/SwE/exfmri

and stepped through the first parts of the example using fmri data.  You will be happy to know that using MATLAB R2018a and SwE toolbox version noted above, that it seems to run fine.

I am still waiting on my researcher about regenerating the SwE.mat file.

To whom should I send updates/suggested corrections for the NISOx page?  There are a couple of suggestions to make it a bit more clear to the uninitiated, and the name of one of the buttons in the SwE interface seems to have changed since it was written.

**Comment by nicholst** [comment]
It is really beyond the call of duty(!) but if you want to make a PR (or
just raise an issue) on the website see here:
https://github.com/NISOx-BDI/NISOx-BDI.github.io/tree/master/Software/SwE

**Comment by TomMaullin** [comment]
Hi @justbennet ,

I can confirm I have now downloaded and looked into your `SwE.mat` file and can confirm you had indeed found an issue that needed resolving! Thank you again for your continued feedback and detailed error reports! Hopefully the error should be resolved shortly and I will let you know once the patch is made available!

@nicholst , I shall have a PR ready for this shortly addressing and explaining the issue in full.

**Comment by TomMaullin** [comment]
This should now be resolved in PR #70, which was merged today. The updated version of the code can now be found in the master branch of this repository.
